packages/xstate-test/src/TestModel.ts

```diff
@@ -102,7 +108,9 @@ export class TestModel<TState, TEvent extends EventObject> {
     let minWeight = Infinity;
     let shortestPaths: Array<TestPath<TState, TEvent>> = [];
 
-    const paths = this.filterPathsTo(statePredicate, this.getShortestPaths());
```

Comment:
The problem was here - paths were deduplicated before the filter (deduplication was happening in `this.getPaths` that was called by `this.getShortestPaths`). So it was possible for the deduplication algorithm/step to remove something that was supposed to be matched by the `statePredicate`